File: ios/Claudiator/Utilities/URLValidator.swift
import Foundation

enum URLValidator {
    /// Cleans and validates a server URL string.
    /// Returns the cleaned URL string, or nil if invalid.
    static func cleanAndValidate(_ input: String) -> String? {
        var url = input.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !url.isEmpty else { return nil }
        if url.hasSuffix("/") { url.removeLast() }
        if !url.hasPrefix("http") {
            let isLocal = url.contains(".local") || url.starts(with: "localhost") || url.starts(with: "127.0.0.1")
            url = (isLocal ? "http://" : "https://") + url
        }
        guard URL(string: url) != nil else { return nil }
        return url
    }
}

File: android/app/src/main/java/com/claudiator/app/util/URLValidator.kt
package com.claudiator.app.util

import java.net.URI

object URLValidator {
    fun cleanAndValidate(input: String): String? {
        var url = input.trim()
        if (url.isEmpty()) return null
        if (url.endsWith("/")) url = url.dropLast(1)
        if (!url.startsWith("http")) {
            val isLocal = url.contains(".local") ||
                url.startsWith("localhost") ||
                url.startsWith("127.0.0.1")
            url = (if (isLocal) "http://" else "https://") + url
        }
        return try {
            URI(url).toURL()
            url
        } catch (_: Exception) {
            null
        }
    }
}
